crytic_compile/compiler/compiler.py

```diff
@@ -37,16 +42,21 @@ def look_for_installed_version(self) -> None:
         This function queries solc-select to see if the current compiler version is installed
         And if its not it will install it
 
-        Returns:
-
+        Raises:
+            UnsupportedSolcVersion: If the version requested cannot be automatically installed by solc-select.
         """
 
         # pylint: disable=import-outside-toplevel
         try:
             from solc_select import solc_select
 
-            if self.version not in solc_select.installed_versions():
+            if (
+                self.version not in solc_select.installed_versions()
+                and self.version in solc_select.get_available_versions()
```

Comment:
`get_available_versions` makes an url query. `install_artifacts` calls also `get_available_versions`, so here we end up making two time the same URL query, which is not great.

Maybe we can rather update solc-select to return boolean in `install_artifacts` to indicate success?

Reply:
Looks like `install_artifacts` returns True on success as of [this PR to solc-select](https://github.com/crytic/solc-select/pull/108) and, as before, throws on failure. Given the throw on failure, we could `try`/`catch` the `install_artifacts` call instead of using an `if` statement to save the extra url query.

Reply:
I think that was an oversight. We should return false https://github.com/crytic/solc-select/issues/143. I also opened issues to cache these requests https://github.com/crytic/solc-select/issues/145.